### PR TITLE
Drop the Foolscap dependency in allmydata.stats

### PR DESCRIPTION
**Pull request by exarkun**
Fixes https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3883

### CODE REVIEW

**Comment by itamarst** [comment]
I tried reviewing this and I am too sleep deprived to understand logic, apparently. Will try again later in week if someone else doesn't get to this.

File: src/allmydata/stats.py [hunk 1]
```diff
@@ -1,54 +1,38 @@
...
         service.MultiService.__init__(self)
-        # we don't use process_time() here, because the constructor is run by
-        # the twistd parent process (as it loads the .tac file), whereas the
-        # rest of the program will be run by the child process, after twistd
-        # forks. Instead, set self.initial_cpu as soon as the reactor starts
-        # up.
-        self.initial_cpu = 0.0 # just in case
-        eventually(self._set_initial_cpu)
-        self.samples = []
+        self.samples: list[tuple[float, float]] = deque([], self.HISTORY_LENGTH + 1)
```
Comment (itamarst) [inline src/allmydata/stats.py:24]: Huh, didn't know `deque` had a length limit option.

You should be using `from typing import Deque, Tuple; Deque[Tuple[float, float]]` apparently.

File: src/allmydata/stats.py [hunk 2]
```diff
@@ -1,54 +1,38 @@
...
 
 from allmydata.util import log, dictutil
 from allmydata.interfaces import IStatsProducer
 
 @implementer(IStatsProducer)
 class CPUUsageMonitor(service.MultiService):
-    HISTORY_LENGTH = 15
-    POLL_INTERVAL = 60  # type: float
+    HISTORY_LENGTH: int = 15
+    POLL_INTERVAL: float = 60
```
Comment (itamarst) [inline src/allmydata/stats.py:19]: It's not a float?

**Review by itamarst — APPROVED** [review]
Couple minor fixes to type annotations.

File: src/allmydata/stats.py (hunk 2 above)
Comment (exarkun) [reply src/allmydata/stats.py:19]: I *guess* the way this works (but feel free to correct me) is that int is considered a subtype of float for the purposes of type checking so it is allowed and correct to assign both float and int values to a variable declared as of type float.

https://peps.python.org/pep-0483/#covariance-and-contravariance is the most directly relevant documentation I was able to easily dig up on the matter.

**Pull request merged**